src/runtime/hexagon/target/fastrpc/CMakeLists.txt

```diff
@@ -70,81 +67,66 @@ set(TVM_REMOTE_D_SKEL_C "tvm_remote_skel.c")
 set(TVM_REMOTE_D_STUB_C "tvm_remote_stub.c")
 
 add_custom_command(
-    OUTPUT ${TVM_REMOTE_D_SKEL_C} ${TVM_REMOTE_D_STUB_C}
-           "${FASTRPC_SRC}/include/${TVM_REMOTE_D_H}"
-    COMMAND ${QAIC_EXE} ${QAIC_FLAGS}
-            "${FASTRPC_SRC}/include/${TVM_REMOTE_D_IDL}"
-    COMMAND ${CMAKE_COMMAND} -E rename "${TVM_REMOTE_D_H}"
-                "${FASTRPC_SRC}/include/${TVM_REMOTE_D_H}"
-    MAIN_DEPENDENCY "${FASTRPC_SRC}/include/${TVM_REMOTE_D_IDL}"
+  OUTPUT ${TVM_REMOTE_D_SKEL_C} ${TVM_REMOTE_D_STUB_C}
+         "${FASTRPC_SRC}/include/${TVM_REMOTE_D_H}"
+  COMMAND ${QAIC_EXE} ${QAIC_FLAGS}
+          "${FASTRPC_SRC}/include/${TVM_REMOTE_D_IDL}"
+  COMMAND ${CMAKE_COMMAND} -E rename "${TVM_REMOTE_D_H}"
+          "${FASTRPC_SRC}/include/${TVM_REMOTE_D_H}"
+  MAIN_DEPENDENCY "${FASTRPC_SRC}/include/${TVM_REMOTE_D_IDL}"
 )
 
 
 if("${FASTRPC_LIBS}" STREQUAL "SKEL")
   # Skel libraries.
   #
-  set(HEXARCH_DIR_v60 "ADSPv60MP")
-  set(HEXARCH_DIR_v62 "ADSPv62MP")
-  set(HEXARCH_DIR_v65 "computev65")
-  set(HEXARCH_DIR_v66 "computev66")
-  set(HEXARCH_DIR_STR "HEXARCH_DIR_${HEXAGON_ARCH}")
-  set(HEXARCH_DIR ${${HEXARCH_DIR_STR}})
-
-  if(NOT HEXARCH_DIR)
-    message(SEND_ERROR
-            "Please set HEXAGON_ARCH to one of v60, v62, v65, v66")
-  endif()
-
-  include_directories(
-      SYSTEM ${HEXAGON_SDK_ROOT}/libs/common/qurt/${HEXARCH_DIR}/include/qurt)
-  include_directories(
-      SYSTEM ${HEXAGON_SDK_ROOT}/libs/common/qurt/${HEXARCH_DIR}/include/posix)
+  include_directories(SYSTEM ${HEXAGON_QURT_INCLUDES})
 
   # Extra compile flags (both C and C++).
   set(EXTRA_COMP_FLAGS
     "-O3"
     "-m${HEXAGON_ARCH}"
   )
   string(REGEX REPLACE ";" " " EXTRA_COMP_FLAGS_STR "${EXTRA_COMP_FLAGS}")
-  message(STATUS "EXTRA_COMP_FLAGS_STR: ${EXTRA_COMP_FLAGS_STR}")
   set(CMAKE_C_FLAGS "${EXTRA_COMP_FLAGS_STR} ${CMAKE_C_FLAGS}")
   set(CMAKE_CXX_FLAGS "${EXTRA_COMP_FLAGS_STR} ${CMAKE_CXX_FLAGS}")
 
   set(EXTRA_LINK_FLAGS
-      "-Wl,--no-threads"
-      "-Wl,--wrap=malloc"
-      "-Wl,--wrap=calloc"
-      "-Wl,--wrap=free"
-      "-Wl,--wrap=realloc"
-      "-Wl,--wrap=memalign"
-      "-Wl,--wrap=posix_memalign"
-      "-Wl,--wrap=__stack_chk_fail"
+    "-Wl,--no-threads"
+    "-Wl,--wrap=malloc"
+    "-Wl,--wrap=calloc"
+    "-Wl,--wrap=free"
+    "-Wl,--wrap=realloc"
+    "-Wl,--wrap=memalign"
+    "-Wl,--wrap=posix_memalign"
+    "-Wl,--wrap=__stack_chk_fail"
   )
   string(REGEX REPLACE ";" " " EXTRA_LINK_FLAGS_STR "${EXTRA_LINK_FLAGS}")
 
   # Extra linker flags for linking shared libraries.
   set(CMAKE_SHARED_LINKER_FLAGS
-        "${EXTRA_LINK_FLAGS_STR} ${CMAKE_SHARED_LINKER_FLAGS}")
+    "${EXTRA_LINK_FLAGS_STR} ${CMAKE_SHARED_LINKER_FLAGS}"
+  )
```

Comment:
Setting global flags like this can be problematic if this project is used by another. Can you set these flags on the specific targets they are needed for. You can use `set_target_properties(libraryname PROPERTIES LINK_FLAGS ${EXTRA_LINK_FLAGS_STR})`.

Reply:
Done.